core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -230,33 +238,36 @@ public void messageReceived(Request request, TransportChannel channel) throws Ex
         }
     }
 
-    class PrimaryOperationTransportHandler implements TransportRequestHandler<Request> {
+    class PrimaryOperationTransportHandler implements TransportRequestHandler<RequestWithAllocationID<Request>> {
         @Override
-        public void messageReceived(final Request request, final TransportChannel channel) throws Exception {
+        public void messageReceived(final RequestWithAllocationID<Request> request, final TransportChannel channel) throws Exception {
             throw new UnsupportedOperationException("the task parameter is required for this operation");
         }
 
         @Override
-        public void messageReceived(Request request, TransportChannel channel, Task task) {
-            new AsyncPrimaryAction(request, channel, (ReplicationTask) task).run();
+        public void messageReceived(RequestWithAllocationID<Request> request, TransportChannel channel, Task task) {
+            new AsyncPrimaryAction(request.request, request.allocationId, channel, (ReplicationTask) task).run();
         }
     }
 
     class AsyncPrimaryAction extends AbstractRunnable implements ActionListener<PrimaryShardReference> {
 
         private final Request request;
+        /** allocationId of the shard this request is meant for */
+        private final String allocationId;
         private final TransportChannel channel;
         private final ReplicationTask replicationTask;
 
-        AsyncPrimaryAction(Request request, TransportChannel channel, ReplicationTask replicationTask) {
+        AsyncPrimaryAction(Request request, String allocationId, TransportChannel channel, ReplicationTask replicationTask) {
```

Comment:
indeed. Fixed.
> On 05 Sep 2016, at 20:55, Jason Tedor <notifications@github.com> wrote:
> 
> In core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java:
> 
> >          private final TransportChannel channel;
> >          private final ReplicationTask replicationTask;
> >  
> > -        AsyncPrimaryAction(Request request, TransportChannel channel, ReplicationTask replicationTask) {
> > +        AsyncPrimaryAction(Request request, String allocationId, TransportChannel channel, ReplicationTask replicationTask) {
> 
> I think you missed this one, and the corresponding field (it looks like you edit the comment on the field though).
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>